client/me/purchases/payment/edit-card-details/index.jsx

```diff
@@ -71,26 +62,21 @@ const EditCardDetails = React.createClass( {
 	 * @param fields
 	 */
 	mergeCard( card, fields: {} ) {
+		const { name } = card;
+
 		return extend( {}, fields, {
-			name: card.name
+			name
 		} );
 	},
 
 	componentWillMount() {
-		const options = {
-			validatorFunction: this.validate,
-			onNewState: this.setFormState
-		};
-
-		if ( this.props.card ) {
-			const fields = formState.createNullFieldValues( this.fieldNames );
-
-			options.initialState = formState.createInitialFormState( this.mergeCard( this.props.card, fields ) );
-		} else {
-			options.fieldNames = this.fieldNames;
-		}
+		const fields = this.mergeCard( this.props.card, formState.createNullFieldValues( this.fieldNames ) );
 
-		this.formStateController = formState.Controller( options );
+		this.formStateController = formState.Controller( {
+			initialState: formState.createInitialFormState( fields ),
```

Comment:
You could get rid of `intialState` and use the `initialFields` property instead: the form state controller will then take care of [creating this initial state](https://github.com/Automattic/wp-calypso/blob/master/client/lib/form-state/index.js#L30).